emukit/bayesian_optimization/acquisitions/max_value_entropy_search.py

```diff
@@ -109,3 +111,184 @@ def evaluate(self, x: np.ndarray) -> np.ndarray:
+
+    def evaluate(self, x: np.ndarray) -> np.ndarray:
+        """
+        Computes the information gain, i.e the change in entropy of p_min if we would evaluate x.
+        :param x: points where the acquisition is evaluated.
+        """
+        if not self._required_parameters_initialized():
+            self.update_parameters()
+
+        # need model predictions at x and at the  x evaluated on the target fidelity (x_target_fidelity)
+        # remove current fidelity index from sample
+        x_target_fidelity = np.delete(x, self.source_idx, axis=1)
+        # Add target fidelity index to sample
+        idx = np.ones((x.shape[0])) * self.target_information_source_index
+        x_target_fidelity = np.insert(x_target_fidelity, self.source_idx, idx, axis=1)
+
+        # get predicted means and variances and pair-wise covariance
+        fmean, fvar = self.model.predict(x)
+        fsd = np.sqrt(fvar)
+        targetmean, targetvar = self.model.gpy_model.predict(x_target_fidelity, include_likelihood=False)
+        targetsd = np.sqrt(targetvar)
+        # faster to do for loop rather than vectorize to avoid unecessary between term covariance calculations
+        covs = [self.model.get_covariance_between_points(x[i].reshape(1, -1), x_target_fidelity[i].reshape(1, -1)) for i in range(0, x.shape[0])]
+        covs = np.array(covs).reshape(-1, 1)
+
+        # convert to pair-wise correlations
+        corrs = covs / (fsd*targetsd)
+        # correct for some potential rounding errors
+        corrs = np.clip(corrs, -1, 1)
+        corrs[x[:, -1] == self.source_idx] = 1
+
+        # calculate moments of extended skew Gaussian distributions
+        gammas = (self.mins-targetmean)/targetsd
+        denom = 1 / (1 - norm.cdf(gammas))
+        # need to account for numerical instability
+        denom[denom == np.inf] = 0
+        means = corrs * (norm.pdf(gammas)) * denom
+        Vars = 1 + corrs * (means) * (gammas - norm.pdf(gammas) * denom)
+        Vars[Vars <= 0] = 0
+
+        # get upper limits for numerical integration
+        upper_lim = means + 8 * np.sqrt(Vars)
+        lower_lim = means - 8 * np.sqrt(Vars)
+
+        # perform numerical integrations
+        approx = np.array([approx_int(corrs[i],
+                                 gammas[i],
+                                 upper_lim[i],
+                                 lower_lim[i]) for i in range(0, corrs.shape[0])]).reshape(-1, 1)
+        return (0.5*np.log(2*np.pi*np.e)-approx).reshape(-1, 1)
+
+
+def approx_int(corr, gamma, upper_lim, lower_lim):
+    # helper function to numerically approx differential entropy of ESG in MUMBO
+    # if corr is 1 then can just calc exactly (like MES)
+    if corr == 1:
+        term1 = 0.5 * np.log(2 * np.pi * np.e)
+        term2 = np.log(1 - norm.cdf(gamma))
+        term3 = gamma * norm.pdf(gamma) / (2 * (1 - norm.cdf(gamma)))
+        return np.mean(term1 + term2 + term3)
+    else:
```

Comment:
Would be nice to have some more details about what's going on in these if/else branches